runtimes/alibi-explain/mlserver_alibi_explain/explainers/black_box_runtime.py

```diff
@@ -83,4 +83,12 @@ def _infer_impl(self, input_data: Union[np.ndarray, List[str]]) -> np.ndarray:
             ssl_verify_path=self.ssl_verify_path,
         )
         # TODO: do we care about more than one output?
-        return NumpyCodec.decode_output(v2_response.outputs[0])
+        decoded = NumpyCodec.decode_output(v2_response.outputs[0])
+
+        if decoded.ndim > 1 and decoded.shape[-1] == 1:
```

Comment:
is it possible to add a test case?

Reply:
Apologies, browser didn't refresh reviews until I pushed the merged button :disappointed: 

I think this bit should be covered by some of the tests we've got there (not the exact case, but close enough).

Reply:
I am actually curious what the existing alibi runtime tests didnt capture this issue?

Reply:
What the tests don't seem to capture is cases where the return shape is like `[N, 1]` . These are cases where the ML framework (e.g. sklearn) would return just `[N]`, but the recent change in the codecs makes the data-dimensionality explicit. 

In these cases, Alibi seemed to run into issues when getting the prediction back from the model (as it would return a np array with shape `[N, 1]`). 

Just to clarify, this is still not captured by the tests. It would be great to add one, but don't have the capacity now (still fixing other things).